electron/launcher/main-process/helpers/login.js

```diff
@@ -0,0 +1,203 @@
+    .then(userInfo => {
+      resolve({body: body, userInfo: userInfo});
+    })
+    .catch(error => {
+      reject(error);
+    });
+  });
+}
+
+
+
+
+
+
+// AUTO LOGIN
+const autoLogin = () => {
+  return new Promise((resolve, reject) => {
+    // READ AND DECRYPT user_info FILE
+    fs.readFile(path.join(app.getPath('userData'), 'user_info'), 'utf-8').then(encryptedData => {
+      let decrypt = data => {
+        return new Promise((resolve, reject) => {
+          let encrypted = Buffer.from(data, 'hex');
+          let decipher = crypto.createDecipheriv(secret.algorithm, Buffer.from(secret.key, 'hex'), Buffer.from(secret.iv, 'hex'));
+          let decrypted = decipher.update(encrypted);
+          decrypted = Buffer.concat([decrypted, decipher.final()]);
+          resolve(decrypted.toString());
+        })
+      }
+
+      return decrypt(encryptedData).then(userInfo => {
+        return JSON.parse(userInfo);
+      })
+    })
+    // GET USER'S PASSWORD
+    .then(userInfo => {
+      return keytar.getPassword('Dietstory', userInfo.name).then(password => {
+        return aes256.decrypt(password).then(decryptedPassword => {
+          return {...userInfo, password: decryptedPassword};
+        })
+      })
+      .catch(error => {
+        throw error;
+      });
+    })
+    // GET SESSION TOKEN
+    .then(userInfo => {
+      // SESSION IS EXPIRED, MUST RE-LOG
+      if(Date.now() > userInfo.session_expiry){
+        return login({username: userInfo.name, password: userInfo.password}).then(response => {
+          return keytar.getPassword('Dietstory_Session', response.userInfo.name).then(sessionToken => {
+            return {...response.userInfo, password: userInfo.password, session_token: sessionToken};
+          });
+        })
+        .catch(error => {
+          throw error;
+        });
+      }
+      // SESSION IS STILL GOOD, RETRIEVE AND USE IT
+      else{
+        return keytar.getPassword('Dietstory_Session', userInfo.name).then(sessionToken => {
+          if(sessionToken !== null){
+            return {...userInfo, session_token: sessionToken};
+          }
+          else{
+            throw new Error('Cannot find session token');
+          }
+        });
+      }
+    })
+    // GET CSRF TOKEN
+    .then(userInfo => {
+      return keytar.getPassword('Dietstory_CSRF', userInfo.name).then(csrfToken => {
+        if(csrfToken !== null){
+          return {...userInfo, csrf_token: csrfToken};
+        }
+        else{
+          throw new Error('Cannot find CSRF token');
+        }
+      });
+    })
+    .then(userInfo => {
+      sessionKey = userInfo.session_token;      // GLOBAL VARIABLE
+      sessionExpiry = userInfo.session_expiry;  // GLOBAL VARIABLE
+      csrfToken = userInfo.csrf_token;          // GLOBAL VARIABLE
+
+      resolve({
+        name: userInfo.name,
+        email: userInfo.email,
+        is_active: userInfo.is_active,
+        is_superuser: userInfo.is_superuser
+      });
```

Comment:
how come our response here is different from `login` response? Are autoLogin and Login not supposed to return the same responses?

Reply:
The response from `autoLogin()` is exactly the same as the response from Django server when logging in normally.